ttnn/cpp/ttnn/operations/pool/maxpool/max_pool2d.cpp

```diff
@@ -35,8 +44,14 @@ Tensor MaxPool2DOp::invoke(uint8_t queue_id, const Tensor& input_tensor, uint32_
 
     if (!out_memory_config.shard_spec.has_value()) {
         // Input is not sharded. Perform sharding.
+        applied_shard_scheme = applied_shard_scheme == TensorMemoryLayout::INTERLEAVED ? TensorMemoryLayout::HEIGHT_SHARDED : applied_shard_scheme; // default to height sharding
```

Comment:
I think we will want a heurisic to pick best sharding based on tesor dims here instead of just picking HS.
In our ttnn torch integration, compiler passes in just interleaved tensors and we need to make sure that we pick the best variant
to avoid out of memory issues

Reply:
@pavlejosipovic more work for your heuristics :-D Lets do that in a subsequent PR.